web-app/js/portal/filter/Filter.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2015 IMOS
+ *
+ * The AODN/IMOS Portal is distributed under the terms of the GNU General Public License
+ *
+ */
+
+Ext.namespace('Portal.filter');
+
+Portal.filter.Filter = Ext.extend(Object, {
+
+    constructor: function(filterDetail, layer) {
+
+        this.layer = layer;
+        this.filterName = filterDetail.name;
```

Comment:
I agree that all these things can lose the 'filter' prefix.

GeoServer calls them "Layer Name" in its UI. But I like that "id" lets you know this is the unique identifier ("name" doesn't).